x/claim/keeper/claim.go

```diff
@@ -803,3 +803,47 @@ func (k Keeper) DeleteAirdropAndEpoch(ctx sdk.Context, identifier string) error
 	k.epochsKeeper.DeleteEpochInfo(ctx, fmt.Sprintf("airdrop-%s", identifier))
 	return k.SetParams(ctx, params)
 }
+
+func (k Keeper) UpdateAirdropAddress(ctx sdk.Context, existingAddress string, newStrideAddress string, airdropId string) bool {
+	logPrefix := "[CLAIM] UpdateAirdropAddress -"
+	airdrop := k.GetAirdropByIdentifier(ctx, airdropId)
+	if airdrop == nil {
+		k.Logger(ctx).Error(fmt.Sprintf("%s airdrop not found for identifier %s", logPrefix, airdropId))
+		return false
+	}
+
+	// verify that the strideAddress is valid
+	_, err := sdk.AccAddressFromBech32(newStrideAddress)
+	if err != nil {
+		k.Logger(ctx).Error(fmt.Sprintf("%s invalid stride address %s", logPrefix, newStrideAddress))
+		return false
+	}
+
+	// note: existingAccAddress will be a STRIDE address with the same coin type as existingAddress
+	existingAccAddress, _ := sdk.AccAddressFromBech32(existingAddress)
+
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, append([]byte(types.ClaimRecordsStorePrefix), []byte(airdrop.AirdropIdentifier)...))
+	if !prefixStore.Has(existingAccAddress) {
+		k.Logger(ctx).Error(fmt.Sprintf("%s no airdrop found in Store for address %s and airdrop identifier %s", logPrefix, existingAddress, airdrop.AirdropIdentifier))
+		return false
+	}
+
+	bz := prefixStore.Get(existingAccAddress)
+	claimRecord := types.ClaimRecord{}
+	err = proto.Unmarshal(bz, &claimRecord)
+	if err != nil {
+		k.Logger(ctx).Error(fmt.Sprintf("%s error unmarshalling claim record for address %s on airdrop %s", logPrefix, existingAddress, airdropId))
+		return false
+	}
```

Comment:
can you just use `k.GetClaimRecord` for this? We should probably also add a `DeleteClaimRecord` keeper function so that this function can just read

```go
oldClaimRecord, err := k.GetClaimRecord(ctx, existingStrideAddress, airdropId)
if err != nil {
   return errorsmod.Wrapf(err, "unable to find claim record for address %s", existingStrideAddress.String())
}

newClaimRecord := oldClaimRecord
newClaimRecord.Address := newStrideAddress
k.SetClaimRecord(ctx, newClaimRecord)

k.DeleteClaimRecord(ctx, oldClaimRecord)
```

Reply:
Agreed, changed!